### PR TITLE
fix: sync one behind issue

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

- New Features
  - Multi-pass oldest→newest timeline draining with a TimelineDrainer pipeline, backpressure-aware buffering, attachment prefetching, and an incremental processing API.
  - TimelineConfig presets and optional TimelineMetrics for tuning and observability; read-marker persists after successful batches.

- Bug Fixes
  - Read-marker advancement deferred until newest ingested event succeeds.
  - Deterministic ordering for same-timestamp events.

- Documentation
  - Expanded sync docs and rollout/monitoring guidance covering draining, backpressure, metrics, and leak prevention.

- Tests
  - Extensive new/updated tests for ordering, multipass drains, listeners (debounce/overflow), read markers, and edge cases.

- Chores
  - Version bump.

### CODE REVIEW

File: lib/features/sync/matrix/matrix_timeline_listener.dart [hunk 1]
```diff
@@ -122,4 +173,108 @@ class MatrixTimelineListener implements TimelineContext {
...
+    while (start < events.length) {
+      final end = (start + _maxPendingBatchSize).clamp(0, events.length);
+      final window = events.sublist(start, end);
+      start = end;
+
+      final seen = <String>{};
+      final unique = <Event>[
+        for (final e in window)
+          if (seen.add(e.eventId)) e,
+      ];
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/sync/matrix/matrix_timeline_listener.dart:228]: changed

File: lib/features/sync/matrix/timeline.dart [hunk 2]
```diff
@@ -7,29 +7,480 @@ import 'package:lotti/features/sync/matrix/consts.dart';
...
+          } else if (eventId == lastReadEventContextId) {
+            continue;
+          }
+          recalculated.add(event);
+        }
+        if (recalculated.isNotEmpty) {
+          newEvents = recalculated;
+          break;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/sync/matrix/timeline.dart:169]: changed
